csharp/ql/lib/ext/generated/Microsoft.DotNet.PlatformAbstractions.model.yml

```diff
@@ -1,12 +1,16 @@
 # THIS FILE IS AN AUTO-GENERATED MODELS AS DATA FILE. DO NOT EDIT.
 extensions:
+  - addsTo:
+      pack: codeql/csharp-all
+      extensible: summaryModel
+    data:
+      - ["Microsoft.DotNet.PlatformAbstractions", "HashCodeCombiner", False, "Add<TValue>", "(TValue,System.Collections.Generic.IEqualityComparer<TValue>)", "", "Argument[0]", "Argument[1]", "taint", "df-generated"]
```

Comment:
I'd expect an `Argument[0]` to `Argument[this]` summary here.

Reply:
The model generator considers simple types as sanitisers. The backing field containing the hash code is a `long`, so we shouldn't expect the `Argument[0] -> Argument[this]` model.
However, the reason we are getting the `Argument[0] -> Argument[1]` (bad) model is for the same reason that we get the `IEqualityComparer<TValue>.Compare` models (there exists an implementation that we can dispatch to, which is considered to manipulate the state of `this` for the `IEqualityComparer`).